distributed/tensor_parallelism/two_d_parallel_example.py

```diff
@@ -52,76 +50,121 @@
 More details can be seen in the slide:
 https://docs.google.com/presentation/d/17g6WqrO00rP3MsxbRENsPpjrlSkwiA_QB4r93_eB5is/
 """
+def find_multiple(n: int, k: int) -> int:
+    """ function to find resizing multiple for SwiGLU MLP """
+    if n % k == 0:
+        return n
+    return n + k - (n % k)
+
+
+class MLP_swiglu(nn.Module):
+    """ SwiGLU to showcase a Llama style MLP model """
+
+    def __init__(self, mlp_dim: int= 1024) -> None:
+        super().__init__()
+        hidden_dim = 4 * mlp_dim
+        scaled_hidden = int(2 * hidden_dim / 3)
+        rounded_hidden = find_multiple(scaled_hidden, 256)
+
+        self.in_proj = nn.Linear(mlp_dim, rounded_hidden, bias=False)
+        self.gate_proj = nn.Linear(mlp_dim, rounded_hidden, bias=False)
+        self.out_proj = nn.Linear(rounded_hidden, mlp_dim, bias=False)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = F.silu(self.in_proj(x)) * self.gate_proj(x)
+        x = self.out_proj(x)
+        return x
+
+"""
+Main body of the demo of a basic version of tensor parallel by using
+PyTorch native APIs.
+"""
+tp_size = 2
+
+
+# understand world topology
+_rank = int(os.environ["RANK"])
+_local_rank = int(os.environ["LOCAL_RANK"])
```

Comment:
why we need to have local_rank and rank defined here? 
"understand world topology", I feel this is sth why we designed a "DeviceMesh" concept, for user to understand the world topology, we should just use DeviceMesh or print DeviceMesh instead

Reply:
removed.  now using mesh.get_rank() for _rank.